services/middlewareServices/FormSGService.ts

```diff
@@ -0,0 +1,57 @@
+import FormSG from "@opengovsg/formsg-sdk"
+import { Request, Response, NextFunction } from "express"
+
+import logger from "@logger/logger"
+
+import { AuthError } from "@root/errors/AuthError"
+
+const formsg = FormSG()
+
+export default class FormSGService {
+  authenticate = () => (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): void => {
+    const signature = req.get("X-FormSG-Signature")
+    if (!signature) {
+      throw new AuthError("Signature missing")
+    }
+    const postUri = `https://${req.get("host")}${req.baseUrl}${req.path}`
+    try {
+      formsg.webhooks.authenticate(signature, postUri)
+      // Continue processing the POST body
+      next()
+    } catch (e) {
+      logger?.error(e)
```

Comment:
do we need the conditional here?

Reply:
https://github.com/isomerpages/isomercms-backend/pull/458/commits/292486b5631d525c3d3d9189758957c50c8c5dad